astropy/modeling/_fitting_parallel.py

```diff
@@ -121,21 +121,18 @@ def _fit_models_to_chunk(
 
     iterating_shape_chunk = data.shape[: len(iterating_axes)]
 
+    model_i = model.copy()
+
     for index in np.ndindex(iterating_shape_chunk):
         # If all data values are NaN, just set parameters to NaN and move on
         if np.all(np.isnan(data[index])):
             for ipar, name in enumerate(model.param_names):
                 parameters[(ipar,) + index] = np.nan
             continue
 
-        # Make a copy of the reference model and inject parameters
-        model_i = _copy_with_new_parameters(
-            model,
-            {
-                name: parameters[(ipar,) + index]
-                for ipar, name in enumerate(model.param_names)
-            },
-        )
+        # Inject parameters into model
+        for ipar, name in enumerate(model.param_names):
```

Comment:
`_copy_with_new_parameters` does a copy and then does
```
    model_new._reset_parameters(**parameters)
```
Can one re-use that or is that also very slow?

Reply:
No that's fine, but setting up the dictionary would take an extra line so wouldn't actually be simpler.

Reply:
Basically it would be something like:

```python
       model_i._reset_parameters(
           **{
                name: parameters[(ipar,) + index]
                for ipar, name in enumerate(model.param_names)
            },
        )
```

so more verbose, hence why I kept it simpler.

Reply:
OK, the only advantage would be that one would have the implementation in just one place. E.g., if, say, some caching were implemented, `_reset_parameters` might get a cache clearing operation, but this might be forgotten. Anyway, not a big deal - the main idea, of moving the copy out of the loop, definitely makes a lot of sense!

Reply:
Ok I went with your suggestion, thanks!